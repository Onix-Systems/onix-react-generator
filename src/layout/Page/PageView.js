import React from 'react';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';
import Header from '../Header/Header';
import styles from './sass/Page.module.scss';

export default function PageView({ children }) {
  const { t } = useTranslation();
  return (
    <div className={styles.app}>
      <Header
        home={t('home')}
        about={t('about')}
        message={t('welcome')}
      />
      {children}
    </div>
  );
}

PageView.propTypes = {
  children: PropTypes.element.isRequired
};
