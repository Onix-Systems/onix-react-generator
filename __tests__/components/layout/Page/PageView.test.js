import React from 'react';
import { render } from '../../../settings/test-utils';
import PageView from '../../../../src/layout/Page/PageView';

describe('<PageView />', () => {
  let wrapper;
  const mockTitle = 'mock text';

  beforeEach(() => {
    wrapper = render(<PageView><h1>{mockTitle}</h1></PageView>);
  });

  test('should render PageView component', () => {
    const { asFragment } = wrapper;
    expect(asFragment(wrapper)).toMatchSnapshot();
  });

  test('PageView should contains children component', () => {
    const { getByText } = wrapper;
    expect(getByText(mockTitle)).toBeInTheDocument();
  });
});
