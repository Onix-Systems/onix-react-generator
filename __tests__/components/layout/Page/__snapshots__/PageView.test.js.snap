// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageView /> should render PageView component 1`] = `
<DocumentFragment>
  <div
    class="app"
  >
    <header
      class="appHeader"
    >
      <h1
        class="appTitle"
      >
        Try to build something great
      </h1>
      <a
        class="headerLink"
        href="/"
      >
        Home
      </a>
      <a
        class="headerLink"
        href="/about"
      >
        About
      </a>
      <button
        class="toggleLanguage"
        data-testid="toggle-language-button"
        type="button"
      >
        en
      </button>
    </header>
    <h1>
      mock text
    </h1>
  </div>
</DocumentFragment>
`;
